diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1acb75a9..05198910 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -215,12 +215,7 @@ if(NOT USE_SYSTEM_MARISA)
   message(STATUS "Use bundled marisa library.")
   add_subdirectory(deps/marisa-0.2.6)
 else()
-  find_library(LIBMARISA NAMES marisa)
-  if (LIBMARISA)
-    message(STATUS "libmarisa found: ${LIBMARISA}")
-  else()
-    message(FATAL_ERROR "libmarisa not found.")
-  endif()
+  find_package(Marisa CONFIG REQUIRED)
 endif()
 
 ######## Subdirectories
